pkg/cloudprovider/aws/packing/packing.go

```diff
@@ -63,104 +54,41 @@ func NewPacker(ec2 ec2iface.EC2API) Packer {
 	var packings []*Packing
 	var packing *Packing
-	var err error
-	remainingPods := constraints.Pods
+	remainingPods := pods
+	nodeCapacities := []*nodeCapacity{}
+	for _, instanceTypeInfo := range instanceTypes {
+		nodeCapacities = append(nodeCapacities, nodeCapacityFrom(*instanceTypeInfo))
+	}
 	for len(remainingPods) > 0 {
-		packing, remainingPods, err = p.packWithLargestPod(remainingPods, constraints)
-		if err != nil {
-			return packings, fmt.Errorf("packing with largest pod, %w", err)
-		}
+		packing, remainingPods = packWithLargestPod(remainingPods, nodeCapacities)
 		// checked all instance type and found no packing option
 		if len(packing.Pods) == 0 {
 			zap.S().Warnf("Failed to find instance type for pod %s/%s ", remainingPods[0].Namespace, remainingPods[0].Name)
 			remainingPods = remainingPods[1:]
 			continue
 		}
 		packings = append(packings, packing)
-		zap.S().Debugf("For %d pod(s) instance types selected are %v", len(packing.Pods), packing.InstanceTypes)
-	}
-	return packings, nil
-}
-
-func (p *podPacker) getNodeCapacities(constraints *cloudprovider.Constraints) ([]*nodeCapacity, error) {
-	result := []*nodeCapacity{}
-
-	describeInstanceTypesInput := &ec2.DescribeInstanceTypesInput{
-		Filters: describeInstanceTypesFiltersFrom(constraints),
-	}
-
-	err := p.ec2.DescribeInstanceTypesPagesWithContext(context.TODO(), describeInstanceTypesInput, func(page *ec2.DescribeInstanceTypesOutput, lastPage bool) bool {
-		for _, instanceTypeInfo := range page.InstanceTypes {
-			nc := nodeCapacityFrom(*instanceTypeInfo)
-			kubeletOverhead := binpacking.CalculateKubeletOverhead(nc.total)
-			if ok := nc.reserve(resources.Merge(constraints.Overhead, kubeletOverhead)); !ok {
-				zap.S().Infof("Excluding instance type %s because there are not enough resources for the kubelet overhead", nc.instanceType)
-			}
-			result = append(result, nc)
-		}
-		return lastPage
-	})
-
-	if err != nil {
-		return nil, fmt.Errorf("fetching instance types using ec2.DescribeInstanceTypes, %w", err)
-	}
-	return result, nil
-}
-
-func describeInstanceTypesFiltersFrom(constraints *cloudprovider.Constraints) []*ec2.Filter {
-	architecture := "x86_64"
-	if constraints.Architecture != nil && *constraints.Architecture == provisioning.ArchitectureArm64 {
-		architecture = string(*constraints.Architecture)
-	}
-
-	filters := []*ec2.Filter{
-		{
-			Name:   aws.String("processor-info.supported-architecture"),
-			Values: []*string{&architecture},
-		},
-		{
-			Name:   aws.String("supported-usage-class"),
-			Values: []*string{aws.String("on-demand")},
-		},
-		{
-			Name:   aws.String("supported-virtualization-type"),
-			Values: []*string{aws.String("hvm")},
-		},
-		{
-			Name:   aws.String("burstable-performance-supported"),
-			Values: []*string{aws.String("false")},
-		},
+		zap.S().Debugf("For %d pod(s), %d instance types are selected %v", len(packing.Pods), len(packing.InstanceTypes), packing.InstanceTypes)
```

Comment:
I'd write this in active voice, "Selected %d instance types options for %d pod(s)"